src/driver/src/DefaultDriver.cpp

```diff
@@ -55,6 +66,103 @@ YAML::Node loadConfig(const std::string& source,
     }
 }
 
+YAML::Node parseExternal(const YAML::Node& external,
+                         YamlParameters& params,
+                         const fs::path& phaseConfig) {
+    int nodeSize = 1;
+
+    fs::path path(external["Path"].as<std::string>());
+    path = fs::absolute(phaseConfig / path);
+
+    if (!fs::is_regular_file(path)) {
+        auto os = std::ostringstream();
+        os << "Invalid path to external PhaseConfig: " << path
+           << ". Please ensure your workload file is placed in 'workloads/[subdirectory]/' and the "
+              "'Path' parameter is relative to the 'phase_configs/' directory";
+        throw InvalidConfigurationException(os.str());
+    }
+
+    auto replacement = loadConfig(path.string());
+
+    if (external["Parameters"]) {
+        nodeSize++;
+        auto newParams = external["Parameters"].as<YamlParameters>();
+        params.insert(newParams.begin(), newParams.end());
+    }
+
+    if (external["Key"]) {
+        nodeSize++;
+        const auto key = external["Key"].as<std::string>();
+        if (!replacement[key]) {
+            auto os = std::ostringstream();
+            os << "Could not find top-level key: " << key
+               << " in phase config YAML file: " << path.string();
+            throw InvalidConfigurationException(os.str());
+        }
+        replacement = replacement[key];
+    }
+
+    if (external.size() != nodeSize) {
+        throw InvalidConfigurationException(
+            "Invalid keys for 'External'. Please set 'Path' and if any, 'Parameters");
+    }
+
+    return recursiveParse(replacement, params, phaseConfig);
+}
+
+YAML::Node replaceParam(const YAML::Node& input, YamlParameters& params) {
+    if (!input["Name"] || !input["Default"]) {
+        throw InvalidConfigurationException(
+            "Invalid keys for '^Parameter', please set 'Name' and 'Default'");
+    }
+
+    auto name = input["Name"].as<std::string>();
+    // The default value is mandatory.
+    auto defaultVal = input["Default"];
+
+    auto paramVal = params.find(name);
+    YAML::Node out;
+
+    if (paramVal != params.end()) {
+        out = paramVal->second;
+    } else {
+        out = defaultVal;
+    }
+
+    // Nested params are ignored for simplicity.
+    return out;
+}
+
+YAML::Node recursiveParse(const YAML::Node& node,
+                          YamlParameters& params,
+                          const fs::path& phaseConfig) {
+    YAML::Node out;
+    switch (node.Type()) {
+        case YAML::NodeType::Map: {
+            std::string fileName;
+            for (YAML::const_iterator it = node.begin(); it != node.end(); ++it) {
+                if (it->first.as<std::string>() == "^Parameter") {
```

Comment:
any reason to do old-skool versus range-based-for e.g. `for(auto&& kvp : node)`?

Reply:
Nope. Goofballed.